### PR TITLE
Feature liq 1145 primary market

### PR DESCRIPTION
**Pull request by nikagamkrelidze**

### CODE REVIEW

File: src/Lykke.Service.IndexHedgingEngine.Domain/PrimaryMarketHistoryItem.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using System;
+
+namespace Lykke.Service.IndexHedgingEngine.Domain
+{
+    public class PrimaryMarketHistoryItem
```
Comment (aushakov) [inline src/Lykke.Service.IndexHedgingEngine.Domain/PrimaryMarketHistoryItem.cs]: Why do you use different naming for same entities?

File: src/Lykke.Service.IndexHedgingEngine.DomainServices/PrimaryMarket/PrimaryMarketService.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                {
+                    AssetId = x.AssetId,
+                    Balance = x.Balance,
+                    Reserved = x.Reserved
+                })
+                .Where(x => x.Balance != 0 || x.Reserved != 0)
+                .ToArray();
+        }
+
+        public async Task UpdateBalanceAsync(string assetId, decimal amount, string userId, string comment)
```
Comment (aushakov) [inline src/Lykke.Service.IndexHedgingEngine.DomainServices/PrimaryMarket/PrimaryMarketService.cs]: Add info log for update balance operation

**Review by aushakov — CHANGES_REQUESTED** [review]

File: client/Lykke.Service.IndexHedgingEngine.Client/Models/PrimaryMarket/PrimaryMarketBalanceModel.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+using JetBrains.Annotations;
+
+namespace Lykke.Service.IndexHedgingEngine.Client.Models.PrimaryMarket
+{
+    /// <summary>
+    /// Represents the balance item of Primary Market wallet.
+    /// </summary>
+    [PublicAPI]
+    public class PrimaryMarketBalanceModel
```
[the remark this answers is not in the document]
Comment (nikagamkrelidze) [reply client/Lykke.Service.IndexHedgingEngine.Client/Models/PrimaryMarket/PrimaryMarketBalanceModel.cs]: Property "Exchange" doesn't make sense there.

File: client/Lykke.Service.IndexHedgingEngine.Client/Models/PrimaryMarket/PrimaryMarketBalanceChangeModel.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        
+        /// <summary>
+        /// Amount by which the balance should be changed.
+        /// </summary>
+        public decimal Amount { set; get; }
+        
+        /// <summary>
+        /// User that performs the update.
+        /// </summary>
+        public string UserId { set; get; }
```
[the remark this answers is not in the document]
Comment (nikagamkrelidze) [reply client/Lykke.Service.IndexHedgingEngine.Client/Models/PrimaryMarket/PrimaryMarketBalanceChangeModel.cs]: It's a response property, not request.

File: src/Lykke.Service.IndexHedgingEngine.DomainServices/PrimaryMarket/PrimaryMarketService.cs (hunk 2 above)
Comment (nikagamkrelidze) [reply src/Lykke.Service.IndexHedgingEngine.DomainServices/PrimaryMarket/PrimaryMarketService.cs]: Info is logged in ILykkeExchangeService

**Review by aushakov — APPROVED** [review]

**Pull request merged**